src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -937,87 +1174,348 @@ void BrgemmEmitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brgKernel, in
+    size_t n_gprs_to_save = sizeof(gprs_to_save) / sizeof(gprs_to_save[0]);
+
+    h->sub(h->rsp, n_gprs_to_save * gpr_size);
+    for (size_t i = 0; i < n_gprs_to_save; ++i)
+        h->mov(h->ptr[h->rsp + i * gpr_size], gprs_to_save[i]);
+
+    // caller obligation to save k-regs as callee may use them
+    size_t n_k_regs_to_save = 8;
+    h->sub(h->rsp, n_k_regs_to_save * k_mask_size);
+    for (size_t i = 0; i < n_k_regs_to_save; ++i) {
+        if (mayiuse(avx512_core))
+            h->kmovq(h->ptr[h->rsp + i * k_mask_size], Opmask(static_cast<int>(i)));
+        else
+            h->kmovw(h->ptr[h->rsp + i * k_mask_size], Opmask(static_cast<int>(i)));
+    }
+
+    // 1. Caller obligation to save vector registers as callee may use them.
+    // 2. There is an implicit assumption that the host code uses the same
+    // `isa` as the injector. Once the assumption is wrong, `vecs_count` and
+    // `vlen` should be replaced with `host_isa::vlen` and
+    // `host_isa::vecs_count`.
+    h->sub(h->rsp, get_max_vecs_count() * get_vec_length());
+    for (size_t i = 0; i < get_max_vecs_count(); ++i)
+        h->uni_vmovups(h->ptr[h->rsp + i * get_vec_length()], Zmm(i));
+
+    kernel_call(kernel, regs, offsets);
+
     // restore vector registers
     for (int i = static_cast<int>(get_max_vecs_count()) - 1; i >= 0; --i) {
-        h->uni_vmovups(Vmm(i), h->ptr[h->rsp + i * get_vec_length()]);
+        h->uni_vmovups(Zmm(i), h->ptr[h->rsp + i * get_vec_length()]);
     }
     h->add(h->rsp, (get_max_vecs_count()) * get_vec_length());
 
     // restore k registers
-    if (isa == cpu::x64::avx512_core) {
-        for (int i = n_k_regs_to_save - 1; i >= 0; --i) {
-            if (mayiuse(avx512_core))
-                h->kmovq(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
-            else
-                h->kmovw(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
-        }
-        h->add(h->rsp, n_k_regs_to_save * k_mask_size);
+    for (int i = n_k_regs_to_save - 1; i >= 0; --i) {
+        if (mayiuse(avx512_core))
+            h->kmovq(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
+        else
+            h->kmovw(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
     }
+    h->add(h->rsp, n_k_regs_to_save * k_mask_size);
 
     // restore gpr registers
     for (int i = n_gprs_to_save - 1; i >= 0; --i)
         h->mov(gprs_to_save[i], h->ptr[h->rsp + i * gpr_size]);
     h->add(h->rsp, n_gprs_to_save * gpr_size);
 }
 
-void BrgemmEmitter::kernel_execute(const brgemm_kernel_t *brg_kernel, const void *A, const void *B, void *C) {
-    // TODO: There are 4 available abi_params on Windows so we have the copy of brgemm_kernel_execute() function
-    //       with 4 runtime parameters (kernel and I/O) and 4 default parameter values (batch, bs and scratch)
-    brgemm_kernel_params_t brgemm_p;
+void BrgemmCopyBBaseEmitter::execute(matmul::jit_brgemm_matmul_copy_b_t *kernel, const void *src,
```

Comment:
Similarly to `BrgemmEmitter` and `BrgemmWithScratchEmitter`, `BrgemmCopyBEmitter::kernel_call` and `BrgemmCopyBWithCompensationsEmitter::kernel_call` could be merged into one method using just a couple tweaks, and it looks that we should do it.

A more ambitious goal to think about, is whether we can create a wrapper that will be able to emit code to call an arbitrary precompiled binary. It seems that it's completely doable if we add a vector uint64_t values to the kernell_call args. The values will be treated as immediates and will be passed as arguments after the `regs`. We might even need two maps instead of vectors: map{position:pair{reg, offset}} and map{position:imm} to allow for an arbitrary order of reg and imm arguments. What do you think about the idea?